napari/utils/colormaps/colormap.py

```diff
@@ -375,8 +401,18 @@ def map(self, values) -> np.ndarray:
         np.ndarray of same shape as values, but with last dimension of size 4
             Mapped colors.
         """
-        values = np.atleast_1d(values)
-        if values.dtype.kind in {'f', 'U'}:
+        if isinstance(values, np.integer):
+            values = int(values)
+        if isinstance(values, int):
+            if self.use_selection and values != self.selection:
+                return np.array((0, 0, 0, 0))
+            return self.color_dict.get(values, self.default_color)
+        if isinstance(values, (list, tuple)):
+            values = np.array(values)
+        if not isinstance(values, np.ndarray) or values.dtype.kind in {
+            'f',
+            'U',
+        }:
```

Comment:
```suggestion
        if not isinstance(values, np.ndarray) or values.dtype.kind in 'fU':
```

This saves a few lines of code and runs as fast or faster in my testing (and only costs 35ns anyway) (both for values in the set and values not in the set).